generators/entity/index.js

```diff
@@ -1459,7 +1490,11 @@ module.exports = EntityGenerator.extend({
             this.entityClassPlural = pluralize(this.entityClass);
             this.entityInstance = _s.decapitalize(this.name);
             this.entityInstancePlural = pluralize(this.entityInstance);
-            this.entityTableName = _s.underscored(this.name).toLowerCase();
+            if (_.isUndefined(this.tableName)) {
```

Comment:
it is needed but its better done in the constructor, look at my change in master https://github.com/jhipster/generator-jhipster/commit/8d32f72457544a2cf19ea10fcc96d1aa1ad65f7f